packages/postgres-database/tests/test_utils_folders_v2.py

```diff
@@ -0,0 +1,1856 @@
+    role: FolderAccessRole,
+) -> None:
+    result = await connection.execute(
+        sa.select([folders_access_rights.c.folder_id])
+        .where(folders_access_rights.c.folder_id == folder_id)
+        .where(folders_access_rights.c.gid == gid)
+        .where(
+            _get_true_permissions(
+                _get_permissions_from_role(role), folders_access_rights
+            )
+        )
+    )
+    rows = await result.fetchall()
+    assert rows is not None
+    assert len(rows) == 1
+
+
+async def _assert_name_and_description(
+    connection: SAConnection,
+    folder_id: _FolderID,
+    *,
+    name: str,
+    description: str,
+):
+    async with connection.execute(
+        sa.select([folders.c.name, folders.c.description]).where(
+            folders.c.id == folder_id
+        )
+    ) as result_proxy:
+        results = await result_proxy.fetchall()
+        assert results
+        assert len(results) == 1
+        result = results[0]
+        assert result["name"] == name
+        assert result["description"] == description
+
+
+@pytest.fixture
+async def setup_users(
+    connection: SAConnection, create_fake_user: Callable[..., Awaitable[RowProxy]]
+) -> list[RowProxy]:
+    users: list[RowProxy] = []
+    for _ in range(10):
+        users.append(await create_fake_user(connection))  # noqa: PERF401
+    return users
+
+
+@pytest.fixture
+async def setup_users_and_groups(setup_users: list[RowProxy]) -> set[_GroupID]:
+    return {u.primary_gid for u in setup_users}
+
+
+@pytest.fixture
+async def setup_projects_for_users(
+    connection: SAConnection,
+    setup_users: list[RowProxy],
+    create_fake_project: Callable[..., Awaitable[RowProxy]],
+) -> set[_ProjectID]:
+    projects: set[_ProjectID] = set()
+    for user in setup_users:
+        project = await create_fake_project(connection, user)
+        projects.add(project.id)
+    return projects
+
+
+async def test_folder_create(
+    connection: SAConnection, setup_users_and_groups: set[_GroupID]
+):
+    owner_gid = _get_random_gid(setup_users_and_groups)
+
+    # when GID is missing no entries should be present
+    missing_gid = 10202023302
+    await _assert_folder_entires(connection, folder_count=0)
+    with pytest.raises(GroupIdDoesNotExistError):
+        await folder_create(connection, "f1", missing_gid)
+    await _assert_folder_entires(connection, folder_count=0)
+
+    # create a folder ana subfolder of the same name
+    f1_folder_id = await folder_create(connection, "f1", owner_gid)
+    await _assert_folder_entires(connection, folder_count=1)
+    await folder_create(connection, "f1", owner_gid, parent=f1_folder_id)
+    await _assert_folder_entires(connection, folder_count=2)
+
+    # inserting already existing folder fails
+    with pytest.raises(FolderAlreadyExistsError):
+        await folder_create(connection, "f1", owner_gid)
+    await _assert_folder_entires(connection, folder_count=2)
+
+
+async def test__get_top_most_access_rights_entry(
```

Comment:
so you test your private function, that is ok, but the then the naming does not make much sense from here.
what does it mean "top_most_access_rights_entry"??

Reply:
I have properly renamed it now.